pkg/oc/cli/cmd/newapp.go

```diff
@@ -373,6 +379,27 @@ func (o *NewAppOptions) RunNewApp() error {
 				hasMissingRepo = true
 				fmt.Fprintf(out, "%sWARNING: No Docker registry has been configured with the server. Automatic builds and deployments may not function.\n", indent)
 			}
+		case *routeapi.Route:
+			if len(t.Spec.Host) > 0 {
+				containsRoute = true
+				var route *routeapi.Route
+				//check if route processing was completed and host field is prepopulated by router
+				err := wait.PollImmediate(500*time.Millisecond, time.Second*RoutePollTimoutSeconds, func() (bool, error) {
+					route, err = config.OSClient.Routes(t.Namespace).Get(t.Name, metav1.GetOptions{})
+					if err != nil {
+						return false, coreerror.New(fmt.Sprintf("Error while polling route %s", t.Name))
```

Comment:
fyi: fmt.Errorf == errors.New(fmt.Sprintf(...))
but here, just return the error without wrapping it -- you're wrapping it in line 398

Reply:
done